WCFDataService/Client/System/Data/Services/Client/QueryResult.cs

```diff
@@ -740,5 +718,20 @@ private MaterializeAtom CreateMaterializer(ProjectionPlan plan, ODataPayloadKind
                 responseMessageWrapper,
                 payloadKind);
         }
+
+        // $count queries need a way to dispose responseMessage as no enummeration occurs.
```

Comment:
What about LoadPropertyResult which inherits from QueryResult? Seems this dispose() will not be called for LoadProperty...